.github/workflows/fmt-check.yaml

```diff
@@ -0,0 +1,31 @@
+name: Format Check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  fmt-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Check formatting
+        run: |
+          # Run golangci-lint fmt and check if there are any changes
+          make fmt
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "Code is not formatted properly. Please run 'make fmt' and commit the changes."
+            git diff
+            exit 1
+          fi
```

Comment:
You could use the following options


```console
	${GOLANGCI_LINT} fmt -diff ./... 
```

https://github.com/golangci/golangci-lint/blob/2a6a36bfd7ac7bbefe597fee2ac055f9fa3d0837/pkg/commands/fmt.go#L71

Reply:
Nice, if you're interesting in creating follow up PR, please create it. If not, I'll create follow up PR.
What do you think ?